### PR TITLE
Persistence Stripout

### PR DESCRIPTION
**Pull request by jpalac**

### CODE REVIEW

File: src/ServiceControl.Persistence.SqlServer/SqlDbCustomCheckDataStore.cs [hunk 1]
```diff
@@ -115,39 +115,38 @@ public async Task<QueryResult<IList<CustomCheck>>> GetStats(PagingInfo paging, s
...
-                               [OriginatingEndpointHost] nvarchar(300) NOT NULL
-                           ) ON [PRIMARY]
-                       END";
-
-                connection.Open();
-
-                await connection.ExecuteAsync(createCommand).ConfigureAwait(false);
-            }
-        }
+        //public static async Task Setup(string connectionString)
```
Comment (tmasternak) [inline src/ServiceControl.Persistence.SqlServer/SqlDbCustomCheckDataStore.cs:118]: Should this be commented?

File: src/ServiceControl/CustomChecks/CustomChecksComponent.cs [hunk 2]
```diff
@@ -24,29 +21,15 @@ public override void Configure(Settings settings, IHostBuilder hostBuilder)
...
-                        break;
-                    default:
-                        break;
-                }
             });
         }
 
         public override void Setup(Settings settings, IComponentInstallationContext context)
         {
+            //TODO can we do this dynamically somehow?
```
Comment (tmasternak) [inline src/ServiceControl/CustomChecks/CustomChecksComponent.cs:29]: Do we still need this?

File: src/ServiceControl/Infrastructure/Settings/DataStoreType.cs [hunk 3]
```diff
@@ -6,4 +6,11 @@ public enum DataStoreType
         RavenDb = 2,
         SqlDb = 3
     }
+
+    public static class DataStoreConfig
+    {
+        public static string InMemoryPersistence = "ServiceControl.Persistence.InMemory.InMemoryPersistenceConfiguration, ServiceControl.Persistence.InMemory";
```
Comment (tmasternak) [inline src/ServiceControl/Infrastructure/Settings/DataStoreType.cs:12]: Should we change the name of the property to e.g. `InMemoryPersistenceTypeFullyQualifiedName`?

File: src/ServiceControl/SetupBootstrapper.cs [hunk 4]
```diff
@@ -2,7 +2,7 @@ namespace Particular.ServiceControl
 {
     using System;
     using System.Threading.Tasks;
-    using global::ServiceControl.CustomChecks;
+    //using global::ServiceControl.CustomChecks;
```
Comment (tmasternak) [inline src/ServiceControl/SetupBootstrapper.cs:5]: Not needed?

File: src/ServiceControl.Persistence.SqlServer/SqlDbCustomCheckDataStore.cs (hunk 1 above)
Comment (jpalac) [reply src/ServiceControl.Persistence.SqlServer/SqlDbCustomCheckDataStore.cs:118]: It's temporary until I figure out how to best do the setup

File: src/ServiceControl/Infrastructure/Settings/DataStoreType.cs (hunk 3 above)
Comment (jpalac) [reply src/ServiceControl/Infrastructure/Settings/DataStoreType.cs]: I have renamed as suggested

File: src/ServiceControl/CustomChecks/CustomChecksComponent.cs (hunk 2 above)
Comment (jpalac) [reply src/ServiceControl/CustomChecks/CustomChecksComponent.cs:29]: Yes until we agree how to do the setup - currently it's in a separate file as a static function - trying to see if we can make it part of the persistence dll and load/run it dynamically.

File: src/ServiceControl/Bootstrapper.cs [hunk 5]
```diff
@@ -117,12 +121,7 @@ void CreateHost()
                 })
                 .UseLicenseCheck()
                 .UseMetrics(settings.PrintMetrics)
-                .UseEmbeddedRavenDb(context =>
-                {
-                    var documentStore = new EmbeddableDocumentStore();
-                    RavenBootstrapper.Configure(documentStore, settings);
-                    return documentStore;
-                })
+                .UseEmbeddedRavenDb(documentStore)
```
[the remark this answers is not in the document]
Comment (jpalac) [reply src/ServiceControl/Bootstrapper.cs]: Good idea - I have moved it into the `SetupPersistence` extension

File: src/ServiceControl/SetupBootstrapper.cs (hunk 4 above)
Comment (jpalac) [reply src/ServiceControl/SetupBootstrapper.cs]: removed

**Pull request merged**